### DISCUSSION TITLE
Timer

### DISCUSSION THREAD

**Discussion by WLS04** [ANLAB-KAIST/KENSv3#193] [Project Q&A 2. Transport Layer (TCP)] [answered]
While implement PA2, as we do not consider about unreliable case, I think timeout is not needed.
So, I waited for all blocked system call without using timer and passed all reliable test case(Also for another random seed).
./app/kens/testopen.cpp (test-kens-open)
./app/kens/testbind.cpp (test-kens-bind)
./app/kens/testhandshake.cpp (test-kens-handshake)
./app/kens/testclose.cpp (test-kens-close)

But document said we need to use timer in order not to wait indefinitely.
For PA2, if our code passed given reliable testcase, can we allowed to implement timer in PA3, not in PA2?
(Document에는 timeout을 위한 timer 구현에 관한 내용이 있었지만, reliable test case의 경우 이 부분이 필요하지 않다고 생각해서 따로 구현하지 않았는데 test case pass에는 문제가 없는 것 같습니다. Timer를 PA3에서 구현해도 괜찮을까요?)

**Comment by Pusnow** [comment, marked as answer]
Yes, as long as you pass all test cases, it is okay of PA2.
